app/javascript/components/FiltersBoxSearchable.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
+import pluralize from 'pluralize';
+
+
+import { fetchFacetFilters } from 'lib/scp-api';
+import FiltersBox from './FiltersBox';
+import FiltersSearchBar from './FiltersSearchBar';
+
+/**
+ * Component for filter search and filter lists
+ */
+export default function FiltersBoxSearchable(props) {
+
+  // State that is specific to FiltersBox
+  const [matchingFilters, setMatchingFilters] = useState(props.facet.filters);
+  const [hasFilterSearchResults, setHasFilterSearchResults] = useState(false);
+
+  // TODO: Get opinions, perhaps move to a UI code style guide.
+  //
+  // Systematic, predictable IDs help UX research and UI development.
+  //
```

Comment:
Concur.  We also need a good sense of what ids should and shouldn't be used for.  Specifically, I think that we should be looking to minimize as much as possible the use of ids in our javascript handlers--using getElementById is usually a sign that the code isn't as modular as it could be (since ids are the DOM equivalent of global vars).  Ids are probably best reserved for CSS and/or instrumentation

Reply:
Agreed!